src/Itinero/Algorithms/Default/ConvexHull.cs

```diff
@@ -0,0 +1,499 @@
+    {
+        private readonly float[] _lats;
+        private readonly float[] _lons;
+
+        public ConvexHull(float[] lats, float[] lons)
+        {
+            _lats = lats;
+            _lons = lons;
+            Points = new int[lats.Length];
+            for (var i = 0; i < Points.Length; i++)
+            {
+                Points[i] = i;
+            }
+        }
+
+        public int[] Points { get; }
+
+        /// <summary>
+        /// The quickhull algorithm in a convenient coordinate format.
+        /// Contains all you need
+        /// </summary>
+        /// <param name="coors"></param>
+        /// <returns></returns>
+        public static List<Coordinate> Quickhull(List<Coordinate> coors)
+        {
+            var lats = new float[coors.Count];
+            var lons = new float[coors.Count];
+
+            for (var i = 0; i < coors.Count; i++)
+            {
+                lats[i] = coors[i].Latitude;
+                lons[i] = coors[i].Longitude;
+            }
+            
+
+            var cv = new ConvexHull(lats, lons);
+
+            var cutoff = cv.Quickhull();
+
+            var results = new List<Coordinate>();
+            for (var i = 0; i < cutoff; i++)
+            {
+                results.Add(coors[cv.Points[i]]);
+            }
+
+            return results;
+        }
+
+
+        /// <summary>
+        /// Removes all given points from allPoints, and checks if any are in the hull.
+        /// If so, a new hull is calculated.
+        /// </summary>
+        /// <param name="allPoints"></param>
+        /// <param name="hull"></param>
+        /// <param name="pointsToRemove"></param>
+        public static List<Coordinate> RemoveFromHull(List<Coordinate> allPoints, List<Coordinate> hull,
+            List<Coordinate> pointsToRemove)
+        {
+            var hullBreach = false;
+            foreach (var toRemove in pointsToRemove)
+            {
+                allPoints.Remove(toRemove);
+                hullBreach = hullBreach || hull.Remove(toRemove);
+            }
+            // There might be points in allPoints which were not part of the hull, but will be now
+            // For correctness, we have to recalculate from scratch
+            return hullBreach ? Quickhull(allPoints) : hull;
+        }
+
+        /// <summary>
+        /// Checks if newPoint is contained within the hull.
+        /// If not, the element is added to the hull-list, at the appropriate position to maintain a with-the-clock order.
+        /// (If inOrder is disabled, the point is just appended as last element)
+        /// The point of insertion is returned
+        /// </summary>
+        /// <param name="hull"></param>
+        /// <param name="newPoint"></param>
+        /// <param name="inOrder"></param>
+        /// <returns></returns>
+        public static int UpdateHull(List<Coordinate> hull, Coordinate newPoint, bool inOrder = true)
+        {
+            if (PointInPolygon.PointLiesWithin(
```

Comment:
@pietervdvn There is a build error here `PointInPolygon` is not found.

Reply:
Weird. Works on my machine ;)

I had a look at it; I didn't really see why this wouldn't build - especially because PiP should be in the exact same namespace. I'm probably missing something stupid and should learn more about C#

Reply:
I don't think it's a C# thing, I tried to fix it but could find they function `PointInPolygon.PointLiesWithin` anywhere in the code.

Are you sure it's not a file not being committed?

Reply:
Hmm ok now it seems to be there... i'm trying again.

Reply:
The file PointInPolygon was deleted in the develop branch, it was lost in the merge with develop.